fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneMailExampleTest.java

```diff
@@ -0,0 +1,257 @@
+ * the FDB Lucene implementation.
+ *
+ */
+@Tag(Tags.RequiresFDB)
+@Tag(Tags.Slow)
+public class LuceneMailExampleTest extends FDBRecordStoreTestBase {
+    private static final Logger LOGGER = LogManager.getLogger(LuceneMailExampleTest.class);
+
+    // If this is true, messages are distributed to MAIL_BOX_COUNT mailboxes randomly based on Gaussian distribution
+    // If this is false, messages are distributed to the mailboxes according to the MAIL_BOX_DISTRIBUTION
+    private static final boolean USE_GAUSSIAN_DISTRIBUTION = false;
+
+    // Use this if mailbox distribution is randomized based on Gaussian distribution
+    private static final int MAIL_BOX_COUNT = 10;
+
+    // Change this variable to increase the batch size
+    private static final int RECORD_COUNT_BATCH_SIZE = 10;
+
+    // Number of batches to load prior to operation
+    private static final int RECORD_COUNT_NUM_BATCHES = 1;
+
+    private static final String DATA_RESOURCE_DIRECTORY = ".out/resources/mail_test_data";
+
+    // Use this to specify the distribution, the number for each index is the sum of all mailboxes' volumes from beginning to it
+    private static final int[] MAIL_BOX_DISTRIBUTION = new int[] {1, 11, 111, 611, 1611, 4611, 10000};
+
+    private static final Index MAIL_EXAMPLE_INDEX = new Index("email_lucene", concat(
+            function(LuceneFunctionNames.LUCENE_STORED, field("mailbox")),
+            field("header", KeyExpression.FanType.FanOut).nest(function(LuceneFunctionNames.LUCENE_FIELD_NAME,
+                            concat(
+                                    function(LuceneFunctionNames.LUCENE_TEXT, field("value")),
+                                    field("key")))),
+            function(LuceneFunctionNames.LUCENE_STORED, field("flags")),
+            function(LuceneFunctionNames.LUCENE_TEXT, concat(field("body"),
+                    function(LuceneFunctionNames.LUCENE_AUTO_COMPLETE_FIELD_INDEX_OPTIONS, value(LuceneFunctionNames.LuceneFieldIndexOptions.DOCS_AND_FREQS_AND_POSITIONS.name())),
+                    function(LuceneFunctionNames.LUCENE_FULL_TEXT_FIELD_INDEX_OPTIONS, value(LuceneFunctionNames.LuceneFieldIndexOptions.DOCS_AND_FREQS_AND_POSITIONS.name())))),
+            field("isSeen"),
+            field("isDeleted"),
+            field("isExpunged"),
+            field("date"),
+            function(LuceneFunctionNames.LUCENE_TEXT, concat(field("headerAsBlob"),
+                    function(LuceneFunctionNames.LUCENE_AUTO_COMPLETE_FIELD_INDEX_OPTIONS, value(LuceneFunctionNames.LuceneFieldIndexOptions.DOCS_AND_FREQS_AND_POSITIONS.name())),
+                    function(LuceneFunctionNames.LUCENE_FULL_TEXT_FIELD_INDEX_OPTIONS, value(LuceneFunctionNames.LuceneFieldIndexOptions.DOCS_AND_FREQS_AND_POSITIONS.name()))))),
+            LuceneIndexTypes.LUCENE,
+            ImmutableMap.of());
+
+
+    @Test
+    public void testReadPerformance() throws Exception {
+        loadData();
```

Comment:
Calling `loadData()` for every test would work for smaller # of records but not for the larger data loads. Not blocking this PR but reusing the data loaded across tests will enable testing of much larger data sets